backend/RestAPI/Controllers/VideosController.cs

```diff
@@ -33,10 +32,88 @@ public class VideosController : ControllerBase
             _videoService = videoService;
         }
 
-        [HttpGet,Route("all")]
-        public async Task<ActionResult<IEnumerable<Video>>> GetAllVideos()
+        [HttpGet, Route("all")]
+        public async Task<IActionResult> AllVideos()
+        {
+            var userIdClaim = User.FindFirst(x => x.Type == ClaimTypes.NameIdentifier);
+            if (userIdClaim is null)
+            {
+                return NotFound();
+            }
+
+            if (!Guid.TryParse(userIdClaim.Value, out var userId))
+            {
+                return NotFound();
+            }
+
+            var videos = await _videoService.GetUserVideos(userId);
+            List<VideoDto> videoDtos = new List<VideoDto>();
+            videos.ForEach(video =>
+            {
+                videoDtos.Add(new VideoDto()
+                {
+                    Id = video.Id,
+                    Title = video.Title,
+                    Size = video.Size,
+                    UploadDate = video.UploadDate.ToString("yyyy-MM-dd")
+                });
+            });
+
+            return Ok(videoDtos);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetVideo(Guid videoId)
+        {
+            var video = await _videosRepository.GetVideoById(videoId);
```

Comment:
Should we call repositories in our controllers?

Reply:
@kibartas Don't see why not. What do you propose?

Reply:
I do not see any benefits of additionally wrapping the repository into some sort of a new service. We would be creating an unnecessary layer of abstractions. The repository kind of **IS** a service.

Our repos are properly abstracted, and I believe that is enough.

Reply:
Read more if interested:
- https://softwareengineering.stackexchange.com/questions/306890/is-it-bad-practice-that-controller-call-repository-instead-of-service
- https://softwareengineering.stackexchange.com/questions/371405/calling-service-and-repository-layer-from-controller